src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonPropertiesView.ts

```diff
@@ -148,6 +148,33 @@ export class ExecutionPlanComparisonPropertiesView extends ExecutionPlanProperti
 		this.populateTable(columns, tableRows);
 	}
 
+	public expandOrCollapsePropertiesTable(): void {
+		const columns: Slick.Column<Slick.SlickData>[] = this.getPropertyTableColumns();
+
+		let primaryProps = [];
+		let secondaryProps = [];
+		if (this._model.primaryElement?.properties) {
+			primaryProps = this._model.primaryElement.properties;
+		}
+		if (this._model.secondaryElement?.properties) {
+			secondaryProps = this._model.secondaryElement.properties;
+		}
+
+		let tableRows = this.convertPropertiesToTableRows(primaryProps, secondaryProps);
+		tableRows = this.sortPropertiesByDisplayValueEquivalency(tableRows);
+
+		switch (this.propertyRowExpansionState) {
```

Comment:
Why do you need to keep the state? You only ever run this when running the actions so it seems like you can just simply set them all collapsed/expanded and leave it at that.